subsys/bluetooth/mesh/brg_cfg.c

```diff
@@ -36,15 +36,19 @@ static ATOMIC_DEFINE(brg_cfg_flags, BRG_CFG_FLAGS_COUNT);
 static void brg_tbl_compact(void)
```

Comment:
A small optimization could be to pass a starting index to `brg_tbl_compact` since we know the first entry that was changed. Then `brg_tbl_compact` can start moving entries from the first removed one.

Reply:
Considering bridging table entry is 8 byte long which is probably 8-10 cycles in memset, Performance gain here could be masked by a need to have extra code for extra variable in `brg_tbl_netkey_removed_evt()` and other removal functions to track the first removed entry, then pass it to this function. So, I think it is worth leaving this change as is.

Reply:
https://godbolt.org/z/hjePerWYj

```
  ldr r3, [r7, #4]
  lsls r3, r3, #3
  ldr r2, .L8+4
  add r3, r3, r2
  movs r2, #8
  movs r1, #0
  mov r0, r3
  bl memset
```

According to my info it is up to 31 cycles. So, assigning a value to `direction` is definitely much less cycles. as well as storing first index (need one variable for this) and passing it to a function.

Reply:
Skipping first entries allows bypassing this code where we know for sure that it is redundant: https://github.com/zephyrproject-rtos/zephyr/blob/fb6ac019fa9404f2c7862e2cfbe2103040b0e507/subsys/bluetooth/mesh/brg_cfg.c#L42-L45. It as you may see copies the same entry to itself including increment of `j`. To avoid this you need a single variable which you assign to a first index in `brg_tbl_netkey_removed_evt` and in `bt_mesh_brg_cfg_tbl_remove`.

If you for example remove the last, the 10th, entry of the table, this code will copy 9 entries to themselves including incrementing `j`, which is 36-40 cycles just for one entry, or 324-360 cycles for this example.

Reply:
This is not a function that used a lot. Adding and removing entries will be done once in a while so adding this will to gain much in speed.

Reply:
I gave rational reason why it should be done. Coping an object to itself is bad code and should be avoided. I gave examples when it matters. The overhead of storing single variable and passing it to a function is significantly less than overhead the current code creates.

Reply:
Added the first entry, as it was not much extra code.